contribs/gmf/src/dropfile/directive.js

```diff
@@ -0,0 +1,119 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import angular from 'angular';
+import {MessageType} from 'ngeo/message/Message.js';
+import './fileAlert.scss';
+
+/**
+ * @ngInject
+ * @param {import("ngeo/message/Notification.js").MessageNotification} ngeoNotification Ngeo notification
+ * @param {import("gmf/datasource/ExternalDataSourcesManager.js").ExternalDatSourcesManager} gmfExternalDataSourcesManager The manager of external datasources.
+ * @param {angular.gettext.gettextCatalog} gettextCatalog Gettext catalog.
+ * @return {angular.IDirective} The Directive Definition Object.
+ * @ngdoc directive
+ * @ngname gmfFileDropZone
+ */
+const fileDrop = function (gmfExternalDataSourcesManager, ngeoNotification, gettextCatalog) {
+  return {
+    restrict: 'A',
+    template: '<div class="drop-zone-off" id="drop-zone"><p>{{"Drop file here." | translate}}</p></div>',
+    /**
+     * @param {angular.IScope} $scope Scope.
+     * @param {JQuery} element Element.
+     * @param {angular.IAttributes} attrs Attributes.
+     */
+    link: function ($scope, element, attrs) {
+      if ($scope.ctrl.fileDropZoneEnabled !== true) {
+        return;
+      }
+      const processDrag = function () {
+        if (event !== null) {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        if (event.type === 'dragenter' || event.type === 'dragleave') {
+          const dropZone = document.getElementById('drop-zone');
+
+          if (event.type === 'dragenter') {
+            dropZone.classList.remove('drop-zone-off');
+            dropZone.classList.add('drop-zone');
+          }
+          if (event.type === 'dragleave') {
+            dropZone.classList.remove('drop-zone');
+            dropZone.classList.add('drop-zone-off');
+          }
+        }
+      };
+
+      element.bind('dragover', processDrag);
+      element.bind('dragenter', processDrag);
+      element.bind('dragleave', processDrag);
+
+      element.bind('drop', function () {
+        if (event !== null) {
+          event.preventDefault();
+        }
+        const dropZone = document.getElementById('drop-zone');
+        dropZone.classList.remove('drop-zone');
+        dropZone.classList.add('drop-zone-off');
+        const file = event.dataTransfer.files[0];
+        gmfExternalDataSourcesManager.createAndAddDataSourceFromFile(file, (success) => {
+          if (!success) {
+            const div = document.createElement('DIV');
+            div.id = 'file-alert';
+
+            const alertElements = document.getElementsByClassName('gmf-app-map-messages');
```

Comment:
Please speak about this particularity in the description (look for `gmf-app-map-messages` class)